### PR TITLE
Add packages to docs

### PR DESCRIPTION
**Pull request by murraycatto**
- Add documenation for packages and package items
- Add documentation for the package add to cart form 
- Add package variable type

### CODE REVIEW

**Comment by kylebyrne** [comment]
@murraycatto just clocked that we'll also need to document that we have a package variable type on blocks

**Comment by murraycatto** [comment]
Thanks @kylebyrne will add

**Comment by murraycatto** [comment]
@kylebyrne done 👍

File: docs/carts/cart_interactions/add_packages.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+layout: default
+title: Add a package
+parent: Cart interactions
+nav_order: 4
+---
+
+# Add a package to your cart
+
+This form allows you to add a package to your cart. It will create a cart if one does not exist.
+This will add all of the packages included items to the cart.
```
Comment (emilyworrall) [inline docs/carts/cart_interactions/add_packages.md:11]: `packages` -> `package's`

File: docs/theme_architecture/blocks/schema/variables/variable_types/package.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+title: Package
+parent: Variable types
+---
+
+# Package
+
+Renders a select input from which a Creator can select a package from the current companies **listable** packages.
+> Listable packages are packages that are not in a draft state.
+
+Returns a [Package]({% link docs/reference/objects/package.md %}) object.
```
Comment (emilyworrall) [inline docs/theme_architecture/blocks/schema/variables/variable_types/package.md:12]: This link doesn't render correctly

**Review by emilyworrall — COMMENTED** [review]
as far as I can tell this makes sense, but maybe it'd be worth getting someone from the FE team to review?

File: docs/carts/cart_interactions/add_packages.md (hunk 1 above)
Comment (murraycatto) [reply docs/carts/cart_interactions/add_packages.md:11]: Thanks :D

File: docs/theme_architecture/blocks/schema/variables/variable_types/package.md (hunk 2 above)
Comment (murraycatto) [reply docs/theme_architecture/blocks/schema/variables/variable_types/package.md:12]: image
What do you see?

Looks fine to me 👀 ?

File: docs/reference/objects/package.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+The package object represents a package, which is a collection of items that can be purchased together.
+
+<br>
+
+#### Attributes
+
+## `package.id`
+{: .d-inline-block }
+
```
Comment (paulina-a) [inline docs/reference/objects/package.md:20]: Could a label be added here for consistency?

File: docs/reference/objects/package.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+## `package.off_sale_copy`
+{: .d-inline-block }
+string
+{: .label .fs-1 }
+
+The copy to display when the package is off sale.
+
+## `package.state`
+{: .d-inline-block }
+
```
Comment (paulina-a) [inline docs/reference/objects/package.md:74]: Could a label be added here for consistency?

File: docs/reference/objects/package/item.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+---
+layout: default
+title: Package Item
+parent: Objects
```
Comment (paulina-a) [inline docs/reference/objects/package/item.md:4]: Would ideally be nested within Package and have Objects as a grandparent 👴

File: docs/reference/objects/package/item.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+Represents an included item in a [package]({% link docs/reference/objects/package.md %}).
+
+<br>
+
+#### Attributes
+
+## `item.id`
+{: .d-inline-block }
+
```
Comment (paulina-a) [inline docs/reference/objects/package/item.md:20]: Could a label be added here for consistency?

File: docs/carts/cart_interactions/add_packages.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+> If any of the items are sold out or the quanity of items being added is greater than the quantity available, the entire package will not be added to the cart.
+
+## Form fields
+
+| Field name                         | Description                                            |
+| ---------------------------------- | ------------------------------------------------------ |
+| `package[id]`                      | The ID of the package to add to the cart.              |
+| `package[items][][id]`             | The ID of the package item for which to add modifiers. |
+| `package[items][][modifier_ids][]` | The ID of the modifier to add to the cart.             |
```
Comment (paulina-a) [inline docs/carts/cart_interactions/add_packages.md:20]: I like this way of breaking down the fields Murray, going to make a note to do the same on the other pages!

**Review by paulina-a — APPROVED** [review]
Thanks @murraycatto this is really clear - just some nits from me

**Pull request merged**